synthetic_data/generators.py

```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
             )
         return profile
 
-    def synthesize(self, num_samples: int, seed=None, noise_level: float = None):
+    def synthesize(
+        self,
+        num_samples: int,
+        noise_level: float = None,
+    ):
         """Generate synthetic tabular data."""
-        if seed is None:
-            seed = self.seed
-
         if noise_level is None:
             noise_level = self.noise_level
 
-        return make_data_from_report(
-            report=self.profile.report(),
-            n_samples=num_samples,
-            noise_level=noise_level,
-            seed=seed,
-        )
+        if self.is_correlated:
+            return make_data_from_report(
+                report=self.profile.report(),
+                n_samples=num_samples,
+                noise_level=noise_level,
+                seed=self.tabular_generator_seed,
+            )
+        else:
+            self.generate_column_data()
+
+            return generate_dataset(
+                rng=self.rng,
+                columns_to_generate=self.col_data,
+                dataset_length=num_samples,
+            )
+
+    def generate_column_data(self):
```

Comment:
Refactored as a separate function in order to clean up synthesize()

Reply:
I think we may be able to improve this still